iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/compaction/IcebergTableOptimizer.java

```diff
@@ -115,27 +138,28 @@ private List<org.apache.hadoop.hive.common.TableName> getTables() {
     }
   }
 
-  private List<String> getPartitions(org.apache.iceberg.Table icebergTable) {
+  private org.apache.hadoop.hive.ql.metadata.Table getHiveTable(String dbName, String tableName) {
```

Comment:
why not reuse 
````
    Table table = metadataCache.computeIfAbsent(ci.getFullTableName(), () ->
        CompactorUtil.resolveTable(conf, ci.dbname, ci.tableName));
````

Reply:
`MetadataCache` caches objects that are descendants of `TBase`: 
`private Cache<String, TBase> metaCache` it doesn't work for `org.apache.hadoop.hive.ql.metadata.Table` which doesn't extend `TBase`.